### PR TITLE
fix partial comment loss

### PR DESCRIPTION
**Pull request by mrahanjam**
# Fix draft comment loss

## Description
This PR ensures users are aware that their draft comment will be lost upon closing comment box without saving.

## PR Type
- [X] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-17935](https://cdap.atlassian.net/browse/CDAP-17935)

## Test Plan
Manual

### CODE REVIEW

File: app/cdap/components/AbstractWidget/Comment/CommentBox.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ import TextareaAutosize from '@material-ui/core/TextareaAutosize';
 import CommentMenu from 'components/AbstractWidget/Comment/CommentMenu';
 import { humanReadableDate } from 'services/helpers';
 import Markdown from 'components/shared/Markdown';
-import { IconButton } from '@material-ui/core';
+import { ClickAwayListener, IconButton } from '@material-ui/core';
 import CloseIcon from '@material-ui/icons/Close';
 import isObject from 'lodash/isObject';
 import If from 'components/shared/If';
```
Comment (de-lan) [inline app/cdap/components/AbstractWidget/Comment/CommentBox.tsx]: nit: can we also remove the usage of `If` whenever we meet them? Thanks!

**Review by de-lan — APPROVED** [review]
LGTM

File: app/cdap/components/PipelineCanvasActions/PipelineComments.tsx [hunk 2]
```diff
@@ -193,6 +203,7 @@ export function PipelineComments({
                   onDelete={onDelete.bind(null, id)}
                   focus={!comment.content.length}
                   disabled={disabled}
+                  onClickAway={onClickAway}
```
Comment (de-lan) [inline app/cdap/components/PipelineCanvasActions/PipelineComments.tsx:205]: Nice additional fix for the canvas comment box!

File: app/cdap/components/AbstractWidget/Comment/DynamicAnchoredComment.tsx [hunk 3]
```diff
@@ -132,48 +131,47 @@ function DynamicAnchor({
...
-                  onClose={onClose}
-                />
-              );
-            })}
-          </Popper>
-        </If>
-      </div>
-    </ClickAwayListener>
+    <div className={classes.root}>
+      <If condition={!isNil(localAnchorEl)}>
```
Comment (de-lan) [inline app/cdap/components/AbstractWidget/Comment/DynamicAnchoredComment.tsx]: nit: remove `If` if possible

**Review by sumengwang — APPROVED** [review]

**Pull request merged**